include/rocksdb/cache.h

```diff
@@ -535,9 +537,6 @@ class Cache {
   // thread safe and should only be called by the caller holding a reference
   // to each of the handles.
   virtual void WaitAll(std::vector<Handle*>& /*handles*/) {}
-
- private:
-  std::shared_ptr<MemoryAllocator> memory_allocator_;
```

Comment:
OK by me. As long as we are clear in HISTORY.md that the API updates to Cache require some updates to custom implementations, I think it will be fine.